### PR TITLE
make lightd availbe,resolve #3363

### LINKED ISSUE #3363: the abci_query of lightd is not availble
We build a lightd, when use:
```
./bnbcli tx 382915C3610861D8BFF7CF854EE4911ABFC13166CDC783C9792FDD8B0A501BD1  --node tcp://localhost:27147  --chain-id XX
```
it will just panic: `index out of range`

### PR DESCRIPTION
**Pull request by guagualvcha**
Resolve #3363

1."abci_query": rpcserver.NewRPCFunc(c.ABCIQuery, "path,data,prove")
"validators": rpcserver.NewRPCFunc(c.Validators, "height"),
the parameters and function do not match, cause index out of range error.
2. the prove of query is forced to be true, while default option is false.
3. fix the wrong key of merkle

* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: cmd/tendermint/commands/lite.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func init() {
 	LiteCmd.Flags().IntVar(&cacheSize, "cache-size", 10, "Specify the memory trust store cache size")
 }
 
-func ensureAddrHasSchemeOrDefaultToTCP(addr string) (string, error) {
+func EnsureAddrHasSchemeOrDefaultToTCP(addr string) (string, error) {
```
Comment (guagualvcha) [inline cmd/tendermint/commands/lite.go]: Expose this fun for thirtpart user use

File: lite/proxy/wrapper.go [hunk 2]
```diff
@@ -145,6 +145,10 @@ func (w Wrapper) Commit(height *int64) (*ctypes.ResultCommit, error) {
 	return res, err
 }
 
+func (w Wrapper) RegisterOpDecoder(typ string, dec merkle.OpDecoder) {
```
Comment (guagualvcha) [inline lite/proxy/wrapper.go]: Expose this fun for thirtpart user to register more op decoder

**Comment by guagualvcha** [comment]
CI report:`FAIL	github.com/tendermint/tendermint/consensus	78.419s
Exited with code 1` . What i modified won't affect consensus, any one help?

**Comment by guagualvcha** [comment]
@melekes  @ebuchman

File: lite/proxy/proxy.go [hunk 3]
```diff
@@ -66,15 +66,15 @@ func RPCRoutes(c rpcclient.Client) map[string]*rpcserver.RPCFunc {
...
+		"validators": rpcserver.NewRPCFunc(c.Validators, "height"),
 
 		// broadcast API
 		"broadcast_tx_commit": rpcserver.NewRPCFunc(c.BroadcastTxCommit, "tx"),
 		"broadcast_tx_sync":   rpcserver.NewRPCFunc(c.BroadcastTxSync, "tx"),
 		"broadcast_tx_async":  rpcserver.NewRPCFunc(c.BroadcastTxAsync, "tx"),
 
 		// abci API
-		"abci_query": rpcserver.NewRPCFunc(c.ABCIQuery, "path,data,prove"),
+		"abci_query": rpcserver.NewRPCFunc(c.ABCIQuery, "path,data"),
```
Comment (ebuchman) [inline lite/proxy/proxy.go]: shouldn't this match the rpc/core/routes.go? `"abci_query": rpc.NewRPCFunc(ABCIQuery, "path,data,height,prove")` ?

**Pull request merged**